client/web/web.go

```diff
@@ -44,7 +44,19 @@ var authenticationRedirectHTML string
 
 var tmpl *template.Template
 
-var localClient tailscale.LocalClient
+// Server is the backend server for a Tailscale web client.
+type Server struct {
+	devMode bool
```

Comment:
What about exporting these fields, and just letting callers set them directly?  For example, in the tsnet use case, the caller is going to need to set the `*tailscale.LocalClient` to the one associated with the tsnet server.  So something like:

``` go
s := new(tsnet.Server)
lc, _ := s.LocalClient()
ws := &web.Server{
    LocalClient: lc,
}
```

Reply:
thoughts on having `NewServer` accept a `*tailscale.LocalClient` instead?

Reply:
yeah, that could certainly work too.  I suspect we have prior art for both throughout the codebase.  I'm just not sure how long this list of method params would grow.  Maybe not too bad.

Reply:
I was trying to think about the the zero value `web.Server` being usable.  But if LocalClient is a pointer, then that doesn't really work.  So in that case, I guess a New method is clearer in how it should be used.

Reply:
That way if we ever have special stuff in there that callers don't need to provide then we don't have to switch stuff over. Could even make the local client arg optional...